go/vt/vtgate/planbuilder/gen4_planner.go

```diff
@@ -408,9 +473,18 @@ func gen4DeleteStmtPlanner(
 	return newPlanResult(plan.Primitive(), operators.TablesUsed(op)...), nil
 }
 
+func insertUnshardedShortcut(stmt *sqlparser.Insert, ks *vindexes.Keyspace, tables []*vindexes.Table) logicalPlan {
+	eIns := &engine.Insert{}
+	eIns.Keyspace = ks
+	eIns.Table = tables[0]
+	eIns.Opcode = engine.InsertUnsharded
+	eIns.Query = generateQuery(stmt)
+	return &primitiveWrapper{prim: eIns}
```

Comment:
don't we want to use the `insert` type here?

Reply:
good catch. done.